libs/domains/services/feature/src/lib/service-list/service-list-action-bar.tsx

```diff
@@ -0,0 +1,330 @@
+        />
+      ),
+    })
+  }
+
+  const handleRestartAllServices = () =>
+    openModal({
+      content: (
+        <ConfirmationModal
+          verb="restart"
+          impactedRows={restartableServices}
+          selectedRows={selectedRows}
+          onSubmit={async () => {
+            try {
+              await restartAllServices({
+                environment,
+                payload: {
+                  application_ids: restartableServices
+                    .filter(({ serviceType }) => serviceType === 'APPLICATION')
+                    .map(({ id }) => id),
+                  container_ids: restartableServices
+                    .filter(({ serviceType }) => serviceType === 'CONTAINER')
+                    .map(({ id }) => id),
+                  database_ids: restartableServices
+                    .filter(({ serviceType }) => serviceType === 'DATABASE')
+                    .map(({ id }) => id),
+                },
+              })
+              resetRowSelection()
+              closeModal()
+            } catch (error) {
+              console.error(error)
+            }
+          }}
+          onCancel={closeModal}
+        />
+      ),
+    })
+
+  const handleStopAllServices = () =>
+    openModal({
+      content: (
+        <ConfirmationModal
+          verb="stop"
+          impactedRows={stoppableServices}
+          selectedRows={selectedRows}
+          onSubmit={async () => {
+            try {
+              await stopAllServices({
+                environment,
+                payload: {
+                  application_ids: stoppableServices
+                    .filter(({ serviceType }) => serviceType === 'APPLICATION')
+                    .map(({ id }) => id),
+                  container_ids: stoppableServices
+                    .filter(({ serviceType }) => serviceType === 'CONTAINER')
+                    .map(({ id }) => id),
+                  database_ids: stoppableServices
+                    .filter(({ serviceType }) => serviceType === 'DATABASE')
+                    .map(({ id }) => id),
+                  helm_ids: stoppableServices.filter(({ serviceType }) => serviceType === 'HELM').map(({ id }) => id),
+                  job_ids: stoppableServices.filter(({ serviceType }) => serviceType === 'JOB').map(({ id }) => id),
+                },
+              })
+              resetRowSelection()
+              closeModal()
+            } catch (error) {
+              console.error(error)
+            }
+          }}
+          onCancel={closeModal}
+        />
+      ),
+    })
+
+  const handleDeleteAllServices = () =>
+    openModalConfirmation({
+      title: `Delete ${deletableServices.length} ${pluralize(deletableServices.length, 'service')}`,
+      name: 'these services',
+      isDelete: true,
+      action: async () => {
+        try {
+          await deleteAllServices({
+            environment,
+            payload: {
+              application_ids: deletableServices
```

Comment:
could you create a function with to manage the payload? I see you use this several times, I think that would simplify the code